api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java

```diff
@@ -162,4 +205,51 @@ public void fail(@NotNull EconomyException exception) {
      */
     void retrieveHeldCurrencies(@NotNull EconomySubscriber<Collection<UUID>> subscription);
 
+    /**
+     * Request the {@link EconomyTransaction} history, limited by the {@code transactionCount}, of this {@code Account}.
+     *
+     * <p>If the specified {@code transactionCount} is higher than the known transactions, then this will return all the
+     * transactions.
+     * <p>If the specified {@code transactionCount} is {@code -1}, then this will return all the transactions. <i>If you
+     * really need all the transactions, please use this with care, because an economy provider might do a query to a database
+     * .</i>
+     *
+     * @param transactionCount the count of the transactions wanted
+     * @param subscription     the {@link EconomySubscriber} accepting the transaction history
+     * @author MrIvanPlays
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    void retrieveTransactionHistory(
+            int transactionCount, @NotNull EconomySubscriber<Collection<EconomyTransaction>> subscription
+    );
+
+    /**
+     * Request the {@link EconomyTransaction} history, starting from the {@link Temporal} {@code from} specified, of this
+     * {@code Account}.
+     *
+     * @param from         the timestamp to get the transactions from
+     * @param subscription the {@link EconomySubscriber} accepting the transaction history
+     * @author MrIvanPlays
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    default void retrieveTransactionHistory(
```

Comment:
Since we are having the option to specify a `from` could we also specify a `from` and a `to` in another method?

Reply:
done